kahuna/public/js/services/api/leases.js

```diff
@@ -110,20 +128,48 @@ leaseService.factory('leaseService', [
       apiPoll(() => {
         return untilLeasesChange(images);
       }).then(results => {
-        $rootScope.$emit('images-updated', results.map(({image})=>image));
-        $rootScope.$emit('leases-updated');
-
-        return results.map(({leases}) => leases);
+          console.log(results);
+          return results.map(({ image, leases }) => {
+              emitLeaseUpdate(leases);
+              console.log(images, leases);
+                emitImageUpdate(image);
+          });
       });
     }
 
+    const imageUpdates$ = new Subject();
+        imageUpdates$.bufferWithTime(1000).subscribe((images) => {
+            if (images.length == 0) {
+                return;
+            }
+            console.log(images);
+            $rootScope.$emit('images-updated', images);
+        });
+
+    function emitImageUpdate(lease) {
+        imageUpdates$.onNext(lease);
+    }
+    const leaseUpdates$ = new Subject();
+        leaseUpdates$.bufferWithTime(1000).subscribe((leases) => {
+            if (leases.length == 0) { return; }
+            console.log("DUCK", leases, leases.length);
+            $rootScope.$emit('leases-updated');
+    });
+
+    function emitLeaseUpdate(lease) {
+        leaseUpdates$.onNext(lease);
+    }
+
+      //This is a race condition, but I think it's a rare one.
+        //Mitigate by refreshing images with refreshImages before posting.
```

Comment:
Is this work done?

Reply:
Yeah, might not be the best wording but I intended that as a warning to anyone using `untilLeasesChange` in the future.

Reply:
Ah, so `mitigated`?

Reply:
Yup.